### PR TITLE
Model repo labels more concretely

### PR DESCRIPTION
**Pull request by rakuy0**
Output of a review from @Cisphyx. Modeling github issue labels using tags resulted in awkward things like `rep.github.enhancement` when it wasn't really github asserting things, and even getting the user info could result in `rep.myfoouser.bugfix`, which feels odd if `myfoouser` isn't a user in the cortex. Promoting the label to a more concrete element so we can record things like times and which service account applied the label made a bit more sense.

Only thing missing is the service account that added the label, but that's waiting on the big inet:service update.

### CODE REVIEW

File: synapse/models/infotech.py [hunk 1]
```diff
@@ -1365,6 +1371,33 @@ def getModelDefs(self):
                         'doc': 'The ID of the issue in the repository system.',
                     }),
                 )),
+                ('it:dev:label', {}, (
+                    ('id', ('str', {}), {
```
Comment (vEpiphyte) [inline synapse/models/infotech.py:1375]: do we want to strip whitespace?

File: synapse/models/infotech.py [hunk 2]
```diff
@@ -1365,6 +1371,33 @@ def getModelDefs(self):
...
+                        'doc': 'The ID of the label.',
+                    }),
+                    ('title', ('str', {}), {
```
Comment (vEpiphyte) [inline synapse/models/infotech.py:1378]: do we want to strip whitespace?

File: synapse/models/infotech.py [hunk 3]
```diff
@@ -429,6 +432,9 @@ def getModelDefs(self):
                 ('it:dev:repo:issue', ('guid', {}), {
                     'doc': 'An issue raised in a repository.',
                 }),
+                ('it:dev:repo:issue:label', ('guid', {}), {
```
Comment (invisig0th) [inline synapse/models/infotech.py:435]: Lets discuss. I think label probably doesn't need the guid abstraction and could likely be just a `:labels` array on the issue node :+1:

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/models/infotech.py [hunk 4]
```diff
@@ -1366,6 +1372,34 @@ def getModelDefs(self):
                     }),
                 )),
 
+                ('it:dev:label', {}, (
```
Comment (invisig0th) [inline synapse/models/infotech.py:1375]: Ok, i finally grok it :D 

`it:dev:repo:label` -> `it:dev:repo:issue:label`
`it:dev:label` -> `it:dev:repo:label`

final answer ;)

**Review by invisig0th — CHANGES_REQUESTED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**